osu.Game/Online/ScoreDownloadTracker.cs

```diff
@@ -120,29 +98,17 @@ private void attachDownload(ArchiveDownloadRequest<IScoreInfo>? request)
 
         private void onRequestFailure(Exception e) => Schedule(() => attachDownload(null));
 
-        private void itemUpdated(ValueChangedEvent<WeakReference<ScoreInfo>> weakItem)
+        private void itemUpdated(ScoreInfo item) => Schedule(() =>
         {
-            if (weakItem.NewValue.TryGetTarget(out var item))
-            {
-                Schedule(() =>
-                {
-                    if (checkEquality(item, TrackedItem))
-                        UpdateState(DownloadState.LocallyAvailable);
-                });
-            }
-        }
+            if (checkEquality(item, TrackedItem))
+                UpdateState(DownloadState.LocallyAvailable);
+        });
 
-        private void itemRemoved(ValueChangedEvent<WeakReference<ScoreInfo>> weakItem)
+        private void itemRemoved(ScoreInfo item) => Schedule(() =>
         {
-            if (weakItem.NewValue.TryGetTarget(out var item))
-            {
-                Schedule(() =>
-                {
-                    if (checkEquality(item, TrackedItem))
-                        UpdateState(DownloadState.NotDownloaded);
-                });
-            }
-        }
+            if (checkEquality(item, TrackedItem))
+                UpdateState(DownloadState.LocallyAvailable);
```

Comment:
Scary no tests caught this one.
```suggestion
                UpdateState(DownloadState.NotDownloaded);
```

Reply:
no way.. i already triple checked this and fixed it once on this branch before PRing..

Reply:
hold on, how does the old code make sense? surely if the equality check passes, then the score is locally available?

Reply:
oh never mind, didn't notice it's in the removal path

Reply:
Fixed an added test coverage.